src/main/java/com/greenfox/tribesoflagopus/backend/service/BuildingService.java

```diff
@@ -0,0 +1,83 @@
+package com.greenfox.tribesoflagopus.backend.service;
+
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingDto;
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingLevelInputDto;
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingListDto;
+import com.greenfox.tribesoflagopus.backend.model.entity.Building;
+import com.greenfox.tribesoflagopus.backend.model.entity.Kingdom;
+import com.greenfox.tribesoflagopus.backend.repository.BuildingRepository;
+import com.greenfox.tribesoflagopus.backend.repository.KingdomRepository;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BuildingService {
+
+  @Autowired
+  BuildingRepository buildingRepository;
+
+  @Autowired
+  KingdomRepository kingdomRepository;
+
+  @Autowired
+  DtoService dtoService;
+
+  public BuildingListDto createBuildingList(long userId) {
+    List<Building> buildingsToConvertToDto = findKingdomByUserId(userId)
+        .getBuildings();
+    List<BuildingDto> buildingDtos = dtoService.convertFromBuildings(buildingsToConvertToDto);
+    BuildingListDto buildingsToReturn = BuildingListDto.builder()
+        .buildings(buildingDtos)
+        .build();
```

Comment:
This builder could be in your dto service as well, because this one is also related to the dtos.